source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRView.mm

```diff
@@ -272,33 +272,68 @@ - (void)processTextConcurrently:(std::shared_ptr<BaseCardElement> const &)textEl
             } else {
-                std::shared_ptr<MarkDownParser> markDownParser = std::make_shared<MarkDownParser>(textForBlock.c_str());
-                parsedString = [NSString stringWithCString:markDownParser->TransformToHtml().c_str() encoding:NSUTF8StringEncoding];
+                markDownParser = std::make_shared<MarkDownParser>(textForBlock);
             }
+            // MarkDownParser transforms text with MarkDown to a html string
+            parsedString = [NSString stringWithCString:markDownParser->TransformToHtml().c_str() encoding:NSUTF8StringEncoding];
+            NSDictionary *data = nil;
+
+            // use Apple's html rendering only if the string has markdowns
+            if(markDownParser->HasHtmlTags()) {
+                NSString *fontFamilyName = nil;
+                if(!self->_hostConfig.fontFamilyNames){
+                    fontFamilyName = @"'-apple-system',  'HelveticaNeue'";
+                } else {
+                    fontFamilyName = self->_hostConfig.fontFamilyNames[0];
+                }
+                // Font and text size are applied as CSS style by appending it to the html string
+                parsedString = [parsedString stringByAppendingString:[NSString stringWithFormat:@"<style>body{font-family: %@; font-size:%dpx; font-weight: %d;}</style>",
+                                                                      fontFamilyName,
+                                                                      [self->_hostConfig getTextBlockTextSize:textConfigForBlock.size],
+                                                                      [self->_hostConfig getTextBlockFontWeight:textConfigForBlock.weight]]];
 
-            // if correctly initialized, fonFamilyNames array is bigger than zero
-            NSMutableString *fontFamilyName = [[NSMutableString alloc] initWithString:@"'"];
-            [fontFamilyName appendString:[self->_hostConfig.fontFamilyNames componentsJoinedByString:@"', '"]];
-            [fontFamilyName appendString:@"'"];
+                NSData *htmlData = [parsedString dataUsingEncoding:NSUTF16StringEncoding];
+                NSDictionary *options = @{NSDocumentTypeDocumentAttribute: NSHTMLTextDocumentType};
+                data = @{@"html" : htmlData, @"options" : options};
+            } else {
+                int fontweight = [self->_hostConfig getTextBlockFontWeight:textConfigForBlock.weight];
+                // sanity check, 400 is the normal font;
+                if(fontweight <= 0 || fontweight > 900){
+                    fontweight = 400;
+                }
+                UIFont *font = nil;
+                if(!self->_hostConfig.fontFamilyNames){
+		             // normalize font weight to scale of -1.0 to 1.0, and 400 is zero
+                     fontweight -= 400;
+                     if(fontweight > 0) {
+                         fontweight /= 500;
+                     } else {
+                         fontweight /= 300;
+                     }
```

Comment:
The nine CSS values 100..900 should probably map exactly to the nine UIKit values `UIFontWeightUltraLight`..`UIFontWeightBlack`.